src/instructlab/config/show.py

```diff
@@ -14,15 +16,53 @@
 yaml.indent(mapping=2, sequence=4, offset=2)
 
 
+def get_nested_value(data_dict: Dict[str, Any], path: str) -> Any:
+    """Retrieve nested value using dot-separated path."""
+    keys = path.split(".")
+    for key in keys:
+        data_dict = data_dict[key]
+    return data_dict
+
+
 @click.command()
 @click.pass_context
+@click.option(
+    "--json",
+    "-j",
+    "json_format",
+    is_flag=True,
+    help="Output in JSON format instead of YAML.",
+)
+@click.option(
+    "--key",
+    "-k",
+    "key_path",
+    default=None,
+    help="Extract a specific key from the configuration, e.g. -k chat or -k chat.context, also can be used with -j.",
+)
 @clickext.display_params
-def show(ctx: click.Context) -> None:
-    """Displays the current config as YAML"""
-    # TODO: make this use pretty colors like jq/yq
+def show(ctx: click.Context, json_format: bool, key_path: str) -> None:
+    """Displays the current config as YAML or JSON, with an option to extract specific key."""
+
     try:
         commented_map = configuration.config_to_commented_map(ctx.obj.config)
-    except YAMLError as e:
-        click.secho(f"Error loading config as YAML: {e}", fg="red")
+        result = (
+            commented_map if not key_path else get_nested_value(commented_map, key_path)
+        )
+
+        stream = compat.StringIO()
+        yaml.dump(result, stream)
+        output = stream.getvalue()
+
+        if isinstance(result, (dict, list)):
+            if json_format:
+                json_output = json.dumps(result, indent=4)
+                click.echo(json_output)
+            else:
+                click.echo(output)
+        else:
+            click.echo(result)
+
+    except (YAMLError, KeyError) as e:
```

Comment:
Any reason not to separate KeyError here to give a more specific error message?

It might also be better to handle the keyerror more explicitly.  Currently it would print:
```
% ilab config show -k evaluate.invalid
Error loading config or key not found: 'invalid'.
```

I think printing the key up to the error might be more helpful.  In this case:

```
Error loading key: 'evaluate.invalid'
```